src/CoreWCF.Primitives/src/CoreWCF/Description/TypeLoader.cs

```diff
@@ -1117,9 +1122,41 @@ private void VerifyConsistency(OperationConsistencyVerifier verifier)
                     requestDescription.Body.WrapperName = requestDescription.Body.WrapperNamespace = null;
                 }
             }
+
+            if (direction == MessageDirection.Input)
+            {
+                operationDescription.AuthorizeData = new Lazy<ReadOnlyCollection<IAuthorizeData>>(() => GetAuthorizeData(operationDescription).ToList().AsReadOnly());
+            }
+
             return operationDescription;
         }
 
+        private static IEnumerable<IAuthorizeData> GetAuthorizeData(OperationDescription operationDescription)
+        {
+            Type serviceType = typeof(TService);
+            if (serviceType.IsInterface)
+            {
+                using var scope = ServiceProvider.CreateScope();
+                object serviceInstance = scope.ServiceProvider.GetService(serviceType);
```

Comment:
This can still breaks some users. There's one user that I made a fix for where their service type has an IHttpContextAccessor constructor parameter and their app breaks when it can't get one. I don't think this will break them specifically as I think they specify a concrete type to AddService, but if they switched to the interface this would break them.

Reply:
I think I'm okay with saying this feature doesn't work when service type is an interface as long as we can detect that they are using an interface and IAuthorizationCapabilities.SupportsAuthorizationData returns true and either fail startup or output a warning. I'm okay with failing startup and rolling back to a warning if that's a problem as going the other direction is worse developer experience.

Reply:
This is ran lazy so it should work. I wrote a specific unit test to check HttpContextAccessor.HttpContext is not null. please see https://github.com/CoreWCF/CoreWCF/pull/850/commits/bd1fb5fdad760909bd22a5d62b4b8e63bdfb082e

Reply:
Okay, I see that. It's depending on a statically set ServiceProvider which as I mentioned in another comment is fragile and will break. If we had concurrent execution of unit tests (on my todo that we switch to dynamic port usage for all tests so we can) this would break.

Reply:
It also still doesn't use the concrete service type from the InstanceContext which can theoretically change from call to call.

Reply:
@mconnew i pushed https://github.com/CoreWCF/CoreWCF/pull/850/commits/d7481f0c2524680134701bcb7241ddd374e95067. It now relies on InstanceContext instead of ServiceProvider. An AuthorizeData collection is cached per type into `OperationDescription`.